## Lab | Inferential statistics - T-test & P-value

1. We will have another simple example on two sample t test (pooled- when the variances are equal). But this time this is a one sided t-test

In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file files_for_lab/machine.txt. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other

In [25]:
import pandas as pd
import numpy as np
import datetime
import warnings
import matplotlib.pyplot as plt
import seaborn as sns
import math
import scipy.stats as st

* H0: the average new machine packing = average old machine packing 
* H1: the average new machine packing > average old machine packing
* One tailed test
* Significance level 0.05
* DoF = 9

#### Reading data 

In [36]:
df1 = pd.read_table('files_for_lab/machine.txt', encoding = 'utf-16')
df1.columns = ('new_m', 'old_m')
df1


new_m  old_m
0   42.1   42.7
1   41.0   43.6
2   41.3   43.8
3   41.8   43.3
4   42.4   42.5
5   42.8   43.5
6   43.2   43.1
7   42.3   41.7
8   41.8   44.0
9   42.7   44.1

In [37]:
from scipy.stats import ttest_ind, norm
st.ttest_ind(df1['new_m'], df1['old_m'])

Ttest_indResult(statistic=-3.3972307061176026, pvalue=0.0032111425007745158)

CONCLUSION :
* t-statistic = 1,883 > -3.397
* we reject H0

2. An additional problem (not mandatory): In this case we can't assume that the population variances are equal. Hence in this case we cannot pool the variances. Independent random samples of 17 sophomores and 13 juniors attending a large university yield the following data on grade point averages. Data is provided in the file files_for_lab/student_gpa.txt. At the 5% significance level, do the data provide sufficient evidence to conclude that the mean GPAs of sophomores and juniors at the university differ?

In [42]:
df2 = pd.read_table('files_for_lab/student_gpa.txt', encoding = 'utf-8')
df2.columns = ('soph', 'jun')
df2

soph   jun
0   3.04  2.56
1   1.71  2.77
2   3.30  2.70
3   2.88  3.00
4   2.11  2.98
5   2.60  3.47
6   2.92  3.26
7   3.60  3.20
8   2.28  3.19
9   2.82  2.65
10  3.03  3.00
11  3.13  3.39
12  2.86  2.58
13  3.49   NaN
14  3.11   NaN
15  2.13   NaN
16  3.27   NaN

* H0: mean of sophomores = mean of juniors
* H1: mean of sophomores != mean of juniors
* 2 tailed test
* Significance level = 0.05



In [45]:
st.ttest_ind(df2['soph'], df2.dropna()['jun'])

Ttest_indResult(statistic=-0.864325455323425, pvalue=0.39475359666695975)

CONCLUSION :
* t-statistic = 2.201 > -.864
* we reject H0